components/purchase_modal/purchase_modal.tsx

```diff
@@ -112,9 +121,57 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
         this.setState({processing: true, paymentInfoIsValid: false});
     }
 
+    comparePlan = (
+        <a
+            className='ml-1'
+            href='https://mattermost.com/pricing-cloud/'
+            target='_blank'
+            rel='noreferrer'
+            onMouseDown={(e) => {
+                e.preventDefault();
+
+                // MouseDown to track regular + middle clicks
+                trackEvent(
+                    TELEMETRY_CATEGORIES.CLOUD_PURCHASING,
+                    'click_compare_plans',
+                );
+            }}
+        >
+            <FormattedMessage
+                id='cloud_subscribe.contact_support'
+                defaultMessage='Compare plans'
+            />
+        </a>
+    );
+
+    onPlanSelected = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const selectedPlan = findProductInDictionary(this.props.products!, e.target.value);
```

Comment:
Same comment as above regarding the use of the non-null assertion operator on `products`